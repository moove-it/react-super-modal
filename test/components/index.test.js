import React from 'react';
import EnzymeToJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import Modal from '../../src/index';

describe('<Modal /> snapshots', () => {
  const props = {
    children: <div>Im the children</div>,
    isOpen: true,
    onClose: jest.fn(),
  };

  describe('when it\'s open', () => {
    it('renders correctly', () => {
      const subject = mount(<Modal {...props}><div>Im the children</div></Modal>);
      expect(EnzymeToJson(subject)).toMatchSnapshot();
    });
  });

  describe('when it\'s closed', () => {
    it('renders just the base component', () => {
      const newProps = { ...props, isOpen: false };
      const subject = mount(<Modal {...newProps}><div>Im the children</div></Modal>);
      expect(EnzymeToJson(subject)).toMatchSnapshot();
    });
  });

  describe('when it has a setted max width', () => {
    it('renders correctly', () => {
      const newProps = { ...props, maxWidth: '1000px' };
      const subject = mount(<Modal {...newProps}><div>Im the children</div></Modal>);
      expect(EnzymeToJson(subject)).toMatchSnapshot();
    });
  });

  describe('when close button is disabled', () => {
    it('renders correctly', () => {
      const newProps = { ...props, showCloseButton: false };
      const subject = mount(<Modal {...newProps}><div>Im the children</div></Modal>);
      expect(EnzymeToJson(subject)).toMatchSnapshot();
    });
  });
});

describe('<Modal /> unit tests', () => {
  const props = {
    children: <div>Im the children</div>,
    isOpen: true,
    onClose: jest.fn(),
  };

  const subject = mount(<Modal {...props}>
    <div className="children" />
  </Modal>);

  it('renders the children', () => {
    expect(subject.contains(<div className="children" />)).toEqual(true);
  });

  // TODO: do more unit tests! check it contains close button or not when changing props
  // check executing functions of the component, etc.
});
