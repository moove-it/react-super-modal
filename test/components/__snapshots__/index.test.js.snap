// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> snapshots when close button is disabled renders correctly 1`] = `
<Modal
  backgroundColor="#FFFFFF"
  closeButtonColor="#000000"
  closeOnEscapePress={true}
  closeOnOverlayClick={true}
  domElement="body"
  isOpen={true}
  maxWidth="640px"
  onClose={[MockFunction]}
  overlayColor="rgba(0,0,0,0.3)"
  showCloseButton={false}
>
  <div
    className="modal-overlay modal-overlay--animated"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.3)",
      }
    }
  >
    <div
      className="modal-container modal-container--animated"
      onClick={[Function]}
      role="presentation"
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "maxWidth": "640px",
        }
      }
    >
      <div
        className="modal-content"
      >
        <div>
          Im the children
        </div>
      </div>
    </div>
  </div>
</Modal>
`;

exports[`<Modal /> snapshots when it has a setted max width renders correctly 1`] = `
<Modal
  backgroundColor="#FFFFFF"
  closeButtonColor="#000000"
  closeOnEscapePress={true}
  closeOnOverlayClick={true}
  domElement="body"
  isOpen={true}
  maxWidth="1000px"
  onClose={[MockFunction]}
  overlayColor="rgba(0,0,0,0.3)"
  showCloseButton={true}
>
  <div
    className="modal-overlay modal-overlay--animated"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.3)",
      }
    }
  >
    <div
      className="modal-container modal-container--animated"
      onClick={[Function]}
      role="presentation"
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "maxWidth": "1000px",
        }
      }
    >
      <CloseButton
        closeButtonColor="#000000"
        onClose={[MockFunction]}
      >
        <div
          className="close-button"
          onClick={[MockFunction]}
          role="button"
          tabIndex={0}
        >
          <svg
            height="12px"
            version="1.1"
            viewBox="0 0 24 24"
            width="12px"
          >
            <title>
              ic-close
            </title>
            <defs />
            <g
              fill="none"
              fillRule="evenodd"
              id="Icon-Close"
              stroke="none"
              strokeWidth="1"
              transform="translate(-1374.000000, -555.000000)"
            >
              <polygon
                fill="#000000"
                fillRule="nonzero"
                id="ic-close-white"
                points="1398 557.418057 1395.58286 555 1386 564.586483 1376.41714 555 1374 557.418057 1383.58286 567.00454 1374 576.591023 1376.41714 579.009081 1386 569.422598 1395.58286 579.009081 1398 576.591023 1388.41714 567.00454"
              />
            </g>
          </svg>
        </div>
      </CloseButton>
      <div
        className="modal-content"
      >
        <div>
          Im the children
        </div>
      </div>
    </div>
  </div>
</Modal>
`;

exports[`<Modal /> snapshots when it's closed renders just the base component 1`] = `
<Modal
  backgroundColor="#FFFFFF"
  closeButtonColor="#000000"
  closeOnEscapePress={true}
  closeOnOverlayClick={true}
  domElement="body"
  isOpen={false}
  maxWidth="640px"
  onClose={[MockFunction]}
  overlayColor="rgba(0,0,0,0.3)"
  showCloseButton={true}
/>
`;

exports[`<Modal /> snapshots when it's open renders correctly 1`] = `
<Modal
  backgroundColor="#FFFFFF"
  closeButtonColor="#000000"
  closeOnEscapePress={true}
  closeOnOverlayClick={true}
  domElement="body"
  isOpen={true}
  maxWidth="640px"
  onClose={[MockFunction]}
  overlayColor="rgba(0,0,0,0.3)"
  showCloseButton={true}
>
  <div
    className="modal-overlay modal-overlay--animated"
    onClick={[Function]}
    role="presentation"
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.3)",
      }
    }
  >
    <div
      className="modal-container modal-container--animated"
      onClick={[Function]}
      role="presentation"
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "maxWidth": "640px",
        }
      }
    >
      <CloseButton
        closeButtonColor="#000000"
        onClose={[MockFunction]}
      >
        <div
          className="close-button"
          onClick={[MockFunction]}
          role="button"
          tabIndex={0}
        >
          <svg
            height="12px"
            version="1.1"
            viewBox="0 0 24 24"
            width="12px"
          >
            <title>
              ic-close
            </title>
            <defs />
            <g
              fill="none"
              fillRule="evenodd"
              id="Icon-Close"
              stroke="none"
              strokeWidth="1"
              transform="translate(-1374.000000, -555.000000)"
            >
              <polygon
                fill="#000000"
                fillRule="nonzero"
                id="ic-close-white"
                points="1398 557.418057 1395.58286 555 1386 564.586483 1376.41714 555 1374 557.418057 1383.58286 567.00454 1374 576.591023 1376.41714 579.009081 1386 569.422598 1395.58286 579.009081 1398 576.591023 1388.41714 567.00454"
              />
            </g>
          </svg>
        </div>
      </CloseButton>
      <div
        className="modal-content"
      >
        <div>
          Im the children
        </div>
      </div>
    </div>
  </div>
</Modal>
`;
